spec/services/hyrax/multiple_membership_checker_spec.rb

```diff
@@ -40,64 +40,65 @@
-      let(:collection) { create(:collection) }
+      let(:collection) { create(:collection, collection_type: collection_type) }
       let(:collections) { [collection] }
       let(:collection_ids) { collections.map(&:id) }
 
-      before do
-        allow(collection).to receive(:collection_type_gid).and_return(collection_type.gid)
-      end
-
       it 'returns nil' do
         expect(checker).to receive(:single_membership_collections).with(collection_ids).once.and_call_original
-        expect(Collection).to receive(:where).once.and_return(collections)
+        expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: [collection_type.gid]).once.and_return(collections)
         expect(subject).to be nil
       end
     end
 
     context 'when multiple single-membership collection instances are in the list, not including current members' do
-      let(:collection1) { create(:collection, title: ['Foo']) }
-      let(:collection2) { create(:collection, title: ['Bar']) }
+      let(:collection1) { create(:collection, title: ['Foo'], collection_type: collection_type) }
+      let(:collection2) { create(:collection, title: ['Bar'], collection_type: collection_type) }
       let(:collections) { [collection1, collection2] }
       let(:collection_ids) { collections.map(&:id) }
 
-      before do
-        allow(collection1).to receive(:collection_type_gid).and_return(collection_type.gid)
-        allow(collection2).to receive(:collection_type_gid).and_return(collection_type.gid)
-      end
-
-      it 'returns nil' do
+      it 'returns an error' do
         expect(item).not_to receive(:member_of_collection_ids)
         expect(checker).to receive(:single_membership_collections).with(collection_ids).once.and_call_original
-        expect(Collection).to receive(:where).once.and_return(collections)
+        expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: [collection_type.gid]).once.and_return(collections)
         expect(subject).to eq 'Error: You have specified more than one of the same single-membership collection types: Greedy (Foo and Bar)'
       end
+
+      context 'with multiple single membership collection types' do
+        let!(:collection_type_2) { create(:collection_type, title: 'Doc', allow_multiple_membership: false) }
+        let(:collection_types) { [collection_type.gid, collection_type_2.gid] }
+
+        it 'returns an error' do
+          expect(item).not_to receive(:member_of_collection_ids)
+          expect(checker).to receive(:single_membership_collections).with(collection_ids).once.and_call_original
+          expect(Collection).to receive(:where).with(id: collection_ids, collection_type_gid_ssim: collection_types).once.and_call_original
+          expect(subject).to eq 'Error: You have specified more than one of the same single-membership collection types: Greedy (Foo and Bar)'
```

Comment:
Would you mind rewording this to...
```
Error: You have specified more than one collection of the same single-membership collection type: Greedy
```
I wonder about including the collection names.  The tested collections have very short names Foo and Bar.  But typical collections have longer names.

Reply:
I think this should be a separate issue since it's outside of the scope of this PR.  The changes to the tests were just refactoring and fixing and not any functional change.